WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -801,6 +821,11 @@ private extension ProductFormViewController {
         navigationItem.backButtonTitle = viewModel.productModel.name.isNotEmpty ? viewModel.productModel.name : Localization.unnamedProduct
     }
 
+    func updateNavigationBarTitle(title: String) {
+        // Update navigation bar title with variation ID
+        self.title = title
+    }
```

Comment:
I think this would be the correct place to handle the conditional code about what title to set, rather than in `viewDidLoad`. You have access to `viewModel` so you can easily do something like
```swift
guard let variationID = viewModel.productionVariationID else {
    return
}
       
title = Localization.variationViewTitle(variationID: "\(variationID)"))
```

What do you think?

Reply:
You are right. Shall I change it this way?

Reply:
I think so, yes!